File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/achievements/SelectedAchievementModel.ts
import type { AchievementModel } from './AchievementModel';
import type { SelectedAchievementCriteriaModel } from './SelectedAchievementCriteriaModel';

export type SelectedAchievementModel = {
    category: AchievementModel;
    name: string;
    description: string;
    points: number;
    isAccountWide: boolean;
    criteria: SelectedAchievementCriteriaModel;
    nextAchievement: AchievementModel;
    displayOrder: number;
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Achievements/SelectedAchievementModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Achievements;

public class SelectedAchievementModel
{
    [JsonPropertyName("category")]
    public AchievementModel Category { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; }

    [JsonPropertyName("description")]
    public string Description { get; set; }

    [JsonPropertyName("points")]
    public int Points { get; set; }

    [JsonPropertyName("is_account_wide")]
    public bool IsAccountWide { get; set; }

    [JsonPropertyName("criteria")]
    public SelectedAchievementCriteriaModel Criteria { get; set; }

    [JsonPropertyName("next_achievement")]
    public AchievementModel NextAchievement { get; set; }

    [JsonPropertyName("display_order")]
    public int DisplayOrder { get; set; }
}
